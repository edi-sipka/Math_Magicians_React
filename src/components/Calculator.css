* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-size: 100%;
}

.main {
  display: flex;
  justify-content: center;
  align-items: center;
}

.calculator {
  margin: 100px;
  width: 50%;
}

.screen {
  width: 100%;
  height: 60px;
  background: #b5b5bd;
  border: 1px solid #222;
  border: 0;
  color: #fff;
  padding: 10px;
  text-align: right;
  font-size: 24px;
}

.row {
  font-size: 24px;
}

.first-row,
.second-row,
.third-row,
.fourth-row,
.fifth-row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.silver {
  background-color: rgb(161, 161, 161);
}

.color {
  background-color: rgb(235, 86, 0);
  color: #fff;
}

.color:hover {
  background-color: rgb(219, 119, 61);
  color: #fff;
}

.input[type='button'] {
  border: 1px solid #222;
  padding: 10px 20px;
  color: black;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.input[type='button']:active {
  background: #ccc;
}

.expand {
  grid-column: 1 / span 2;
}
