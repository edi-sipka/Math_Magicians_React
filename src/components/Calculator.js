/* eslint-disable */
import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });

  const changeCalculation = (event) => {
    const result = event.target.value;
    const calculation = calculate(state, result);
    setState(calculation);
  };

  const displayResult = (event) => {
    setState({ total: event.target.value });
  };

  const { total, next, operation } = state;
  return (
    <div className="main">
      <h2>Let's do some math!</h2>
      <div className="calculator">
        <div className="screen" onChange={displayResult}>
          {total} {operation} {next}
        </div>
        <div className="row first-row">
          <input value="AC" type="button" onClick={changeCalculation} />
          <input value="+/-" type="button" onClick={changeCalculation} />
          <input value="%" type="button" onClick={changeCalculation} />
          <input
            value="รท"
            type="button"
            className="color"
            onClick={changeCalculation}
          />
        </div>
        <div className="row second-row">
          <input value="7" type="button" onClick={changeCalculation} />
          <input value="8" type="button" onClick={changeCalculation} />
          <input value="9" type="button" onClick={changeCalculation} />
          <input
            value="x"
            type="button"
            className="color"
            onClick={changeCalculation}
          />
        </div>
        <div className="row third-row">
          <input value="4" type="button" onClick={changeCalculation} />
          <input value="5" type="button" onClick={changeCalculation} />
          <input value="6" type="button" onClick={changeCalculation} />
          <input
            value="-"
            type="button"
            className="color"
            onClick={changeCalculation}
          />
        </div>
        <div className="row fourth-row">
          <input value="1" type="button" onClick={changeCalculation} />
          <input value="2" type="button" onClick={changeCalculation} />
          <input value="3" type="button" onClick={changeCalculation} />
          <input
            value="+"
            type="button"
            className="color"
            onClick={changeCalculation}
          />
        </div>
        <div className="row fifth-row">
          <input
            value="0"
            type="button"
            className="expand"
            onClick={changeCalculation}
          />
          <input value="." type="button" onClick={changeCalculation} />
          <input
            value="="
            type="button"
            className="color"
            onClick={changeCalculation}
          />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
