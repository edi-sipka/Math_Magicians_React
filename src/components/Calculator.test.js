/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('Rendering 3', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
