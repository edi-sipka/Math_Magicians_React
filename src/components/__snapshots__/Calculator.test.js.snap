// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering 3 1`] = `
<div
  className="main"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="screen"
      onChange={[Function]}
    >
      0
       
       
    </div>
    <div
      className="row first-row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="color"
        onClick={[Function]}
        type="button"
        value="รท"
      />
    </div>
    <div
      className="row second-row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="color"
        onClick={[Function]}
        type="button"
        value="x"
      />
    </div>
    <div
      className="row third-row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="color"
        onClick={[Function]}
        type="button"
        value="-"
      />
    </div>
    <div
      className="row fourth-row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="color"
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="row fifth-row"
    >
      <input
        className="expand"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="color"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
